hash_practice/exercises.py

```diff
@@ -2,28 +2,84 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
     """
-    pass
+    freq_hash = {}
+
+    for word in strings:
+        key = ''.join(sorted(word))
+        if key not in freq_hash.keys():
+            freq_hash[key] = []
+
+        freq_hash[key].append(word)
+    values_list = []
+    for value in freq_hash.values():
+        values_list.append(value)
+    return values_list
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    if nums == []:
+        return nums
+
+    freq_table = {}
+    for num in nums:
+        if num in freq_table:
+            freq_table[num] += 1
+        else:
+            freq_table[num] = 1
+    max_freq = 0
+    common_elemnts = []
+    for key, value in freq_table.items():
+
+        if freq_table[key] > max_freq:
+            max_freq = freq_table[key]
+            common_elemnts.append(key)
+        elif freq_table[key] == max_freq:
+            common_elemnts.append(key)
+
+    return common_elemnts

```

Comment:
Maybe I'm missing something, but I don't see how this gets only the k most frequent elements.